### ISSUE TITLE
DICOM: What should a DICOM RDF file use as a primary subject?

### ISSUE THREAD

**Issue by ebremer** [w3c/hcls-fhir-rdf#142]
DICOM JSON uses and object {} which contains all of the DICOM properties, essentially a blank node.

For multiple DICOM files or a merged set of DICOM RDF data should this be changed to a URI, a urn:sha256 hash of the source file with owl:sameAs to URIs, or something else?

**Comment by detlef-grittner-sohard** [comment]
The DICOM standard already describes how to uniquely identify data sets with [Unique Identifiers (UIDs)](https://dicom.nema.org/medical/dicom/current/output/chtml/part05/chapter_9.html).

An actual DICOM file or DICOM object typically contains multiple entities and each of these has a UID. There is the [DICOM Model of the Real World](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/chapter_7.html), which can be used as a reference.

A typical DICOM object like, for example, a Computed Tomography Image (CT) has the following entities: Patient, Study, Series, Image. For Study, Series, Image the DICOM standard has Study Instance UID, Series Instance UID, SOP Instance UID, respectively.
Only for Patient there isn't a unique ID, since there is no globally accepted concept how to uniquely identify humans. (In the US there is the SSN, but Germany doesn't have that yet.)

At least for Study, Series, SOP Instance (eg. image data), IRIs can be constructed like:
http://semantic-dicom.org/dcm#Study_{Study_Instance_UID}
http://semantic-dicom.org/dcm#Series_{Series_Instance_UID}
http://semantic-dicom.org/dcm#Image_{SOP_Instance_UID}

If you solve the problem of the Patient ID with a study centric approach like the Study Root Information Model in DICOM, then each study has its own patient and a Patient IRI could be constructed like this:
http://semantic-dicom.org/dcm#Patient_{Study_Instance_UID}

This would allow to merge all instances of entities that according to the DICOM standard always must have the same values for their attributes and circumvent that the wrong patient data is merged by always keeping Patients distinct.

A DICOM file would not contain a single root, but would have instances of entities, which are linked together.

In the [SeDI DICOM ontology from Bioportal ](https://bioportal.bioontology.org/ontologies/SEDI) this is done with verbs derived from the entity relationship model of the DICOM standard, e.g.

http://semantic-dicom.org/dcm#Patient__{Study_Instance_UID} dcm:hasStudy http://semantic-dicom.org/dcm#Series__{Study_Instance_UID} .

As a benefit you can reuse the IRIs outside of a DICOM context to link additional information to Study, Series etc. This inherent ability of RDF is not present in JSON or the binary format of DICOM.

Additional resources:
https://www.iscb.org/cms_addon/conferences/cshals2010/presentations/GrittnerCSHALS10.pdf
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5119276/

**Comment by ebremer** [comment]
@grittner-sohard Would we then map all UI VR types in RDF as URN as in this example [urn:oid:0.9.2342.19200300.100.4](https://www.ietf.org/rfc/rfc3001.txt)?

**Comment by detlef-grittner-sohard** [comment]
@ebremer Yes, that is absolutely possible. The DICOM UIDs are actually OIDs and therefore can be directly inserted into the unr:oid: notation.
For the Patient, which has no defined UID, a generated UUID can be used and then be inserted with the 2.25 root. Which is how the DICOM standard recommends encoding UUIDs ([PS3.5 ChapterB2](https://dicom.nema.org/medical/dicom/current/output/html/part05.html#sect_B.2)).

**Comment by ebremer** [comment]
Using SOP instance UID
```
PREFIX dcm: <https://halcyon.is/dicom/ns/>
PREFIX geo: <http://www.opengis.net/ont/geosparql#>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

<urn:oid:1.3.6.1.4.1.14519.5.2.1.18138783555294854291858006605381019315>
        rdf:type      dcm:SOPInstance;
        dcm:00020001  [ dcm:InlineBinary  "AAE="^^xsd:base64Binary;
                        dcm:vr            "OB"
                      ];
        dcm:00020002  "1.2.840.10008.5.1.4.1.1.481.3";
        dcm:00020003  "1.3.6.1.4.1.14519.5.2.1.18138783555294854291858006605381019315";
        dcm:00020010  "1.2.840.10008.1.2";
        dcm:00020012  "1.3.6.1.4.1.22213.1.143";
        dcm:00020013  "0.5";
...
```

**Comment by detlef-grittner-sohard** [comment]
```
@prefix dcm: <http://semantic-dicom.org/dcm#> .

<urn:oid:1.3.6.1.4.1.14519.5.2.1.21087345762211724523378497892240459677>
    a dcm:Study ;
    dcm:isStudyOf <Patient_1.3.6.1.4.1.14519.5.2.1.21087345762211724523378497892240459677> ;
    dcm:containsSeries <urn:oid:1.3.6.1.4.1.14519.5.2.1.302382790855582445722435410442490497846> ;
    dcm:0020000D "1.3.6.1.4.1.14519.5.2.1.21087345762211724523378497892240459677" .

<Patient_1.3.6.1.4.1.14519.5.2.1.21087345762211724523378497892240459677>
    a dcm:Patient ;
    dcm:hasStudy <urn:oid:1.3.6.1.4.1.14519.5.2.1.21087345762211724523378497892240459677> ;
    dcm:00100020 "Pancreas-CT-CB_001" .

<urn:oid:1.3.6.1.4.1.14519.5.2.1.302382790855582445722435410442490497846>
    a dcm:Series ;
    dcm:isSeriesOf <urn:oid:1.3.6.1.4.1.14519.5.2.1.21087345762211724523378497892240459677> ;
    dcm:containsImage <urn:oid:1.3.6.1.4.1.14519.5.2.1.244661575185439401461912739980091955671> ;
    dcm:0020000E "<dcm:ATT0020000E>1.3.6.1.4.1.14519.5.2.1.302382790855582445722435410442490497846" .
```

**Comment by dbooth-boston** [comment]
1/9/2025: Agreed to use SOP instance UID.  See minutes: https://www.w3.org/2025/01/09-hcls-minutes.html
